flavors/benchmark/benchmark.go

```diff
@@ -63,6 +65,8 @@ type Dependencies struct {
 	awsCfgProvider     awslib.ConfigProviderAPI
 	identityProvider   awslib.IdentityProviderGetter
 	kubernetesProvider providers.KubernetesClientGetterAPI
+	// TODO: add gcp provider here?
```

Comment:
Let me know if you want to discuss this new `Dependencies` struct. It's also possible to initialize `Benchmark`s with a different mechanism if it gets too ugly to keep adding dependencies here